doc/API.md

```diff
@@ -146,3 +157,212 @@ Failing requests may be due to the following errors:
+
+```ssh
+POST /users HTTP/1.1
+Content-Type: application/json
+Authorization: Bearer QWxhZGRpbjpPcGVuU2VzYW1l...
+{
+  "email": "user@domain.org"
+}
+```
+### Response
+Successful requests will produce a "201 Created" response with a body containing an activation url:
+```ssh
+HTTP/1.1 201 Created
+Connection: close
+{
+  "activation_url": "/v1/users/InR5cCI6IkpXVCJ"
+}
+```
+
+Failing requests may be due to the following errors:
+* status code 400, errno 101: Invalid email. Missing or malformed email.
+* status code 400, errno 400: Bad request.
+* status code 409, errno 409: Already exists.
+* status code 401, errno 401: Unauthorized. If credentials are not valid.
+
+## GET /users
+Get the list of all registered users.
+
+Only users with admin privileges are able to access this method.
+
+### Request
+Requests must include an authorization header containing a [bearer token](#authentication).
+```ssh
+GET /users HTTP/1.1
+Authorization: Bearer QWxhZGRpbjpPcGVuU2VzYW1l...
+```
+
+### Response
+Successful requests will produce a "200 OK" response with a body containing the list of all registered users:
+```ssh
+HTTP/1.1 200 OK
+Connection: close
+[{
+  "id": "hfkjsIklksadhs",
+  "username": "admin"
+  "email": "admin@domain.org",
+  "is_admin": true,
+  "is_active": true
+}, {
+  "id": "InR5cCI6IkpXVCJ",
+  "username": "pepe"
+  "email": "user@domain.org",
+  "is_admin": false,
+  "is_active": true
+}, {
+  "id": "kjfIsalj5893213",
+  "email": "another_user@domain.org",
+  "is_admin": false,
+  "is_active": false
+}]
+```
+
+Failing requests may be due to the following errors:
+* status code 400, errno 400: Bad request.
+* status code 401, errno 401: Unauthorized. If credentials are not valid.
+
+## GET /users/:id
+Get the information of the user matching the given id.
+
+Only the owner or users with admin privileges are able to access this method.
+
+### Request
+```ssh
+GET /users/InR5cCI6IkpXVCJ HTTP/1.1
+```
+
+### Response
+Successful requests will produce a "200 Ok" response.
+```ssh
+HTTP/1.1 200 Ok
+Connection: close
+{
+  "id": "InR5cCI6IkpXVCJ",
+  "username": "pepe"
+  "email": "user@domain.org",
+  "is_admin": false,
+  "is_active": true
+}
+```
+
+Failing requests may be due to the following errors:
+* status code 400, errno 400: Bad request.
+* status code 401, errno 401: Unauthorized. If credentials are not valid.
+* status code 404, errno 404: Not Found. The user does not exist.
+
+## PUT /users/:id
+Edit the information of the user matching the given id.
+
+Users with non admin privileges are only able to edit their own information.
+
+### Request
+Requests must include an authorization header containing a [bearer token](#authentication).
+
+___Parameters___
+* id - User id.
+* username - User name.
+* password - User password.
+* is_admin - Flag to give or remove admin permissions.
+
+```ssh
+PUT /users/:id HTTP/1.1
+Content-Type: application/json
+Authorization: Bearer QWxhZGRpbjpPcGVuU2VzYW1l...
+{
+  "username": "pepe",
```

Comment:
Ok, here it doesn't appear the id parameter, so we just need to remove it from the documentation.